src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForContainerBase.cs

```diff
@@ -137,38 +118,34 @@ private static void updateNeighborhood(NeighborhoodBuilder neighborhood, string
          return cache;
       }
 
-      private IMoBiCommand addNeighborhoodsToProject(IReadOnlyList<NeighborhoodBuilder> neighborhoods, MoBiSpatialStructure spatialStructure)
-      {
-         if (neighborhoods == null || !neighborhoods.Any()) return new MoBiEmptyCommand();
-         var command = new MoBiMacroCommand
-         {
-            CommandType = AppConstants.Commands.AddCommand,
-            ObjectType = ObjectTypes.Neighborhood,
-            Description = AppConstants.Commands.AddDependentDescription(spatialStructure, ObjectTypes.Neighborhood, ObjectTypes.SpatialStructure)
-         };
+      protected abstract IMoBiCommand AddNeighborhoodsToSpatialStructure(IReadOnlyList<NeighborhoodBuilder> neighborhoods, MoBiSpatialStructure spatialStructure);
 
-         return neighborhoods.Any(existingItem => !addNeighborhood(existingItem, command, spatialStructure)) ? CancelCommand(command) : command;
+      private void registerLoadedIn(IObjectBase deserializedObject)
+      {
+         Context.Register(deserializedObject);
       }
 
-      private bool addNeighborhood(NeighborhoodBuilder neighborhoodBuilder, MoBiMacroCommand command, MoBiSpatialStructure spatialStructure)
+      private static void updateDiagramLayout(IReadOnlyList<IContainer> allImportedContainers, MoBiSpatialStructure importedSpatialStructure, MoBiSpatialStructure spatialStructure)
       {
-         var forbiddenNames = spatialStructure.NeighborhoodsContainer.Children.Select(x => x.Name).Union(AppConstants.UnallowedNames).ToList();
```

Comment:
this was doing all the code from the InteractionTaskForContainer by hand. Obvioulsy forgetting some steps. I am calling it instead